src/patternfly/components/ModalBox/modal-box.scss

```diff
@@ -94,13 +101,31 @@
       margin-right: var(--pf-c-modal-box--c-button--sibling--MarginRight); // Create room for the close button
     }
   }
+
+  > .pf-c-modal-box__actions {
+    .pf-m-2-actions {
+      + * {
+        margin-right: var(--pf-c-modal-box__actions--m-2-actions--sibling--MarginRight);
+      }
+    }
+
+    + * {
+      margin-right: var(--pf-c-modal-box__actions--sibling--MarginRight);
+    }
+  }
 }
 
 .pf-c-modal-box__header {
+  display: flex;
+  flex-direction: column;
   padding-top: var(--pf-c-modal-box__header--PaddingTop);
   padding-right: var(--pf-c-modal-box__header--PaddingRight);
   padding-left: var(--pf-c-modal-box__header--PaddingLeft);
 
+  &.pf-m-row {
```

Comment:
I wonder if this should be an opt-in to restructure the `__header` element.

Something like

```
.pf-c-modal-box__header.pf-m-help {
 display: flex;
 flex-direction: row;
}
```